libclc/ptx-nvidiacl/libspirv/synchronization/barrier.cl

```diff
@@ -9,14 +9,34 @@
 #include <spirv/spirv.h>
 #include <spirv/spirv_types.h>
 
+int __clc_nvvm_reflect_arch();
+
 _CLC_OVERLOAD _CLC_DEF void __spirv_MemoryBarrier(unsigned int memory,
                                                   unsigned int semantics) {
-  if (memory == CrossDevice) {
-    __nvvm_membar_sys();
-  } else if (memory == Device) {
-    __nvvm_membar_gl();
+
+  // for sm_70 and above membar becomes semantically identical to fence.sc.
+  // However sm_70 and above also introduces a lightweight fence.acq_rel that
```

Comment:
Can you link to the NVIDIA docs that show these version changes?

Reply:
Please put that in the comment and I'll be happy with this patch

Reply:
Done